src/pharmpy/model/statements.py

```diff
@@ -367,6 +389,94 @@ def set_dose(self, compartment, dose):
         nx.relabel_nodes(self._g, mapping, copy=False)
         return new_comp
 
+    def replace_dose(self, compartment, old_dose, new_dose):
```

Comment:
I agree that this would make more sense

Reply:
That was precisely the function. To avoid adding another method, your suggestion has been implemented, replacing based on `admid` as an option in `set_dose`